src/usr/local/www/services_ntpd.php

```diff
@@ -59,6 +59,10 @@
 	unset($input_errors);
 	$pconfig = $_POST;
 
+	if (!is_numericint($_POST['ntpmaxpeers']) || ($_POST['ntpmaxpeers'] < 4) || ($_POST['ntpmaxpeers'] > 10)) {
+		$input_errors[] = gettext("Max candidate NTP peers must be a number between 4 and 10");
```

Comment:
Where did you come up with these minimum and maximum numbers? I don't see any mention of limits on the values in the NTP docs or source. The default is 10, but someone could potentially want something higher if it's a completely internal setup.

Reply:
based on this statement With three servers, you have the minimum number of time sources needed to allow ntpd to dectect if one time source is a "falseticker". However ntpd will then be in the position of choosing from the two remaining sources.This configuration provides no redundancy.

**if someone put only 1 pool and set minclock to less than 4 this can lead to trouble.**

With three servers, you have the minimum number of time sources needed to allow ntpd to dectect if one time source is a "falseticker". However ntpd will then be in the position of choosing from the two remaining sources.This configuration provides no redundancy.

With at least four upstream servers, one (or more) can be a "falseticker", or just unreachable, and ntpd will have a sufficient number of sources to choose from.

Specify the maximum number of servers retained by the server discovery schemes. The default is 10, which should typically be changed. This should be an odd number typically two or three more than minclock, plus the number of pool entries. The pool entries must be added as maxclock, but not minclock, also counts the pool entries themselves. 

**For example, tos maxclock 4 with 1 pool lines would keep 3 associations.**

that said i think that maxclock should start from 4 and default set to 5

ok, 10 was my arbitrary upper limit but i think that a limit should be present, there wasn't a knob before but if people start to play with it and set it to 9999 or a random higher value multiplied by all devices out there, it can become a real problem for the ntp pool. maybe we can set to 20 and see if people ask for more? or you tell me

Reply:
I read the same info but I was mostly curious if you had seen a limit mentioned elsewhere that I missed. `4` seems like a sane minimum (Since `minclock` will always be `3`, and it should be higher). `5` is probably the best default since it should be `minclock+2`. The upper bound is less clear. I guess make it `25` for now just for sanity's sake. If someone needs it higher than that, they can patch the source or open a bug report to have it changed.

Since the values are used in multiple places you should probably define the min/max near the top of the file as a variable or constant, and then reference the variables in the later instances. That way if we decide to change them in the future, only one line must be changed.